### ISSUE TITLE
mutating the `data` prop in a universal load affects the data returned from a server load

### ISSUE THREAD

**Issue by eltigerchino** [sveltejs/kit#12236]
### Describe the bug

> I came across an issue on Discord where someone tried using dynamic import() to load a component and pass it to data in the universal load:
> 
> ```js
> export async function load({ data }) {
>   data.component = (await import('$lib/components/Button.svelte')).default
>   return data
>   //     ^ error
> }
> ```
> This threw an error along the lines of "cannot return non-serializable POJOs" since data was being returned by the server load in +page.server.js. Hopefully my contribution to the docs helps point out this lesser-known nuance.

https://github.com/sveltejs/kit/pull/11822#issue-2128689538

### Reproduction

https://stackblitz.com/edit/sveltejs-kit-template-default-qnnygo?file=src%2Froutes%2F%2Bpage.js

### System Info

```Shell
StackBlitz
```

### Severity

annoyance

### Additional Information

We could:

1. Keep the current behaviour where `data` is mutated.
2. Prevent mutations of `data` from affecting the serialization of the server load data.
3. Throw an error telling users not to mutate `data`.

**Comment by dummdidumm** [comment]
I don't think we should change anything here. Mutation is always something to be wary of. And the referenced PR would only solve this for top level entries but not nested ones which may further the confusion.
